libs/datalayout/src/main/java/com/linkedin/openhouse/datalayout/datasource/TableFileStats.java

```diff
@@ -15,15 +19,39 @@ public class TableFileStats implements DataSource<FileStat> {
 
   @Override
   public Dataset<FileStat> get() {
-    return spark
-        .sql(String.format("SELECT file_path, file_size_in_bytes FROM %s.data_files", tableName))
-        .map(new FileStatMapper(), Encoders.bean(FileStat.class));
+    StructType fileSchema =
+        spark.sql(String.format("SELECT * FROM %s.data_files", tableName)).schema();
+    try {
+      fileSchema.apply("partition");
+      return spark
+          .sql(
+              String.format(
+                  "SELECT file_path, file_size_in_bytes, partition FROM %s.data_files", tableName))
+          .map(new FileStatMapper(), Encoders.bean(FileStat.class));
+    } catch (IllegalArgumentException e) {
+      return spark
+          .sql(
+              String.format(
+                  "SELECT file_path, file_size_in_bytes, null FROM %s.data_files", tableName))
+          .map(new FileStatMapper(), Encoders.bean(FileStat.class));
```

Comment:
Interesting, why does this happen, is this the only way to handle this?

Reply:
If it is not a partition table. We cannot assume the table will be partitioned in this component.